nasa and hhmi hubs: update QGIS image

consideRatio opened the pull request:
Bumps to the image built via https://github.com/2i2c-org/nasa-qgis-image/pull/10.

- Fixes https://github.com/2i2c-org/meta/issues/931
- Removes a workaround to set `singleuser.cmd: null`
  This can be done as the docker image, based on quay.io/jupyter/minimal-notebook, since 2024-01-22 declares an ENTRYPOINT that includes executing the `start.sh` script.
  
  A Dockerfile's `ENTRYPOINT` / `CMD` can be overridden by specifying `command` / `args` in a k8s pod's container specification. The z2jh charts `singleuser.cmd` maps to `KubeSpawner.cmd`, and that configuration together with `KubeSpawner.args` is combined to set k8s pod's container `args`. This means KubeSpawner doesn't touch `ENTRYPOINT` aka. `command` in k8s specifications, making the `start.sh` script still be executed, while when it was set in jupyter/docker-stacks images `CMD` before 2024-01-22 it would get overridden.

yuvipanda commented:
@consideRatio can you help me understand why `cmd: null` is not needed anymore? It's primarily set for https://github.com/2i2c-org/infrastructure/issues/3434, and that particular image requires `start.sh` to run (due to https://github.com/2i2c-org/nasa-qgis-image/blob/babb0fa15dfa5149016a019be595dffc483b08f8/Dockerfile#L14).

yuvipanda commented:
If this is based on https://github.com/jupyter/docker-stacks/pull/2087#event-11551562606, my understanding is that that shouldn't affect us here because Dockerfile's `ENTRYPOINT` is actually Kubernetes' `cmd` and Dockerfile's `CMD` is actually Kubernetes `args`.

consideRatio commented:
Hehe yes but what is kubespawners cmd? :) kubespawners cmd and kubespawners args combines into setting k8s args on the pod i think / from mobile

consideRatio commented:
I've updated the PR description to clarify this further @yuvipanda, and I've also updated the image tags. I consider this PR ready to be merged from a technical standpoint, but we are interfering with the communities user environment which we have no procedure around.

Do you think its OK to bump these images @yuvipanda?

consideRatio commented:
Verified my understanding that kubespawner `cmd` maps to k8s `args`:

```
kubectl get pod -n redacted jupyter-redacted -o yaml | grep -E -A5 "command|args"

  - args:
    - jupyterhub-singleuser
    env:
    - name: CPU_GUARANTEE
      value: "0.05"
    - name: JPY_API_TOKEN
--
  - command:
    - sh
    - -c
    - id && chown 1000:1000 /home/jovyan /home/jovyan/shared && ls -lhd /home/jovyan
    image: busybox:1.36.1
    imagePullPolicy: IfNotPresent
--
  - command:
    - iptables
    - --append
    - OUTPUT
    - --protocol
    - tcp
```

yuvipanda commented:
Ah, somewhere somehow my understanding needs to be rethought, but we don't need to wait for that to merge this. I tested this on staging and it works. Thanks for thinking this through, @consideRatio.

consideRatio commented:
Thank you @yuvipanda!!

I checked that repo2docker based images also doesn't have an issue with setting `singleuser.cmd` which led to #3807.
